File: data/CodeLlama/readable_format_HPEC/fortran/file259.f90
subroutine print_array(a, nr, nq, np)
implicit none

DATA_TYPE, dimension(np, nq, nr) :: a
integer :: nr, nq, np
integer :: i, j, k
do i = 1, nr
do j = 1, nq
do k = 1, np
write(0, DATA_PRINTF_MODIFIER) a(k, j, i)
if (mod((i - 1), 20) &
== 0) then
write(0, *)
end if
end do
end do
end do
write(0, *)
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file259.cpp
#include <iostream>
#define DATA_TYPE int
#define DATA_PRINTF_MODIFIER "%d"
void print_array(DATA_TYPE a[100][100][100], int nr, int nq, int np)
{
    for(int i=0;i<nr;i++)
        for(int j=0;j<nq;j++)
            for(int k=0;k<np;k++)
            {
                printf(DATA_PRINTF_MODIFIER, a[k][j][i]);
                if ((i % 20) == 0)
                    printf("\n");
            }
    printf("\n");
}
